corgi/tasks/pnc.py

```
@@ -0,0 +1,40 @@
+import logging
+
+import requests
+from celery_singleton import Singleton
+from requests.exceptions import HTTPError
+
+from config.celery import app
+from corgi.collectors.pnc import parse_pnc_sbom
+from corgi.core.models import ProductVariant
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(base=Singleton)
+def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
+    logger.info("Fetching PNC SBOM %s for purl %s", sbom_data["id"], purl)
+    """Fetch a PNC SBOM from sbomer"""
+
+    # Validate the supplied product information
+    try:
+        ProductVariant.objects.get(name=product_data["productVariant"])
+    except ProductVariant.DoesNotExist:
+        logger.warning(
+            "PNC SBOM provided for nonexistant product variant: %s", product_data["productVariant"]
+        )
+        return
```

I get that we don't want to be spammed with errors when SBOMer sends messages about variants / products we haven't loaded yet. But hiding the error here means we won't notice if every single message refers to a nonexistent variant.

I guess it's fine (or at least I can't think of a better solution), but let's keep an eye on things after we merge this code. That way we'll know if we process any messages correctly, or if e.g. the variant names in the message are always different than the variant names in ET / Corgi.

---

We can probably ingest the Components without linking to a ProductVariant. Maybe a good idea here to be to make sure any product_data["productVariant"] existing in ET by checking CollectorErrataProductVariant exists by that name?